File: server/aoReporting/Views/OnInstallClass.vb

Option Strict On
Option Explicit On

Imports Contensive.Addons.Reporting.Controllers
Imports Contensive.BaseClasses

Namespace Views

    '
    Public Class OnInstallClass
        Inherits AddonBaseClass
        '
        '=====================================================================================
        ''' <summary>
        ''' 
        ''' </summary>
        ''' <param name="CP"></param>
        ''' <returns></returns>
        Public Overrides Function Execute(ByVal CP As CPBaseClass) As Object
            Dim result As String = ""
            Dim sw As New Stopwatch : sw.Start()
            Try
                '
                ' -- initialize application. If authentication needed and not login page, pass true
                Using ae As New applicationController(CP, False)
                    '
                    ' -- delete the old harcoded email drop report
                    CP.Db.ExecuteNonQuery("delete from ccmenuentries where linkpage='?af=12&rid=28'")
                    '
                    ' -- reset the EmailDropReport
                    Using cs As CPCSBaseClass = CP.CSNew()
                        Dim reportId As Integer = 0
                        If (cs.Open("Admin Framework Reports", "name='Email Drop Report'")) Then
                            reportId = cs.GetInteger("id")
                        End If
                        CP.Db.ExecuteNonQuery("delete from AdminFrameworkReports where id=" & reportId)
                        CP.Db.ExecuteNonQuery("delete from AdminFrameworkReportColumns where reportid=" & reportId)
                    End Using
                End Using
            Catch ex As Exception
                CP.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: server/aoReportingCSharp/AddonsLegacy/OnInstallClass.cs
using Contensive.BaseClasses;
using Contensive.Reporting.Models;
using System;
using System.Diagnostics;

namespace Contensive.Reporting {

    // 
    public class OnInstallClass : AddonBaseClass {
        // 
        // =====================================================================================
        /// <summary>
        ///         ''' 
        ///         ''' </summary>
        ///         ''' <param name="CP"></param>
        ///         ''' <returns></returns>
        public override object Execute(CPBaseClass CP) {
            string result = "";
            Stopwatch sw = new Stopwatch(); sw.Start();
            try {
                // 
                // -- initialize application. If authentication needed and not login page, pass true
                using (ApplicationModel ae = new ApplicationModel(CP, false)) {
                    // 
                    // -- delete the old harcoded email drop report
                    CP.Db.ExecuteNonQuery("delete from ccmenuentries where linkpage='?af=12&rid=28'");
                    // 
                    // -- reset the EmailDropReport
                    using (CPCSBaseClass cs = CP.CSNew()) {
                        int reportId = 0;
                        if ((cs.Open("Admin Framework Reports", "name='Email Drop Report'")))
                            reportId = cs.GetInteger("id");
                        CP.Db.ExecuteNonQuery("delete from AdminFrameworkReports where id=" + reportId);
                        CP.Db.ExecuteNonQuery("delete from AdminFrameworkReportColumns where reportid=" + reportId);
                    }
                }
            } catch (Exception ex) {
                CP.Site.ErrorReport(ex);
            }
            return result;
        }
    }
}
